pkg/pipeline/templates/terraformed.go.tmpl

```diff
@@ -60,3 +61,14 @@ func (tr *{{ .CRD.Kind }}) SetParameters(params map[string]interface{}) error {
 	}
 	return json.TFParser.Unmarshal(p, &tr.Spec.ForProvider)
 }
+
+// GetConnectionDetails of this {{ .CRD.Kind }}
+func (tr *{{ .CRD.Kind }}) GetConnectionDetails(data []byte) (map[string][]byte, error) {
```

Comment:
I remember we switched using `map[string]interface{}` in other terraformed functions but now I am questioning whether it was the right decision. At least, for some functions (e.g. `SetObservation`, `SetParameters` and this one here), we need serialized data and what we do for them is 

`serialized (tfstate) -> unmarshall (map[string]interface) -> marshall -> unmarshall (to spec and status)`
instead of 
`serialized (tfstate) -> unmarshall (to spec and status)` 

Other than concerns like performance and some possibility of data loss, I think this is unnecessary. I would suggest using whichever makes sense (`serialized` vs `map[string]interface`) instead of forcing `map[string]interface `everywhere. 
@muvaf wdyt?

Reply:
Ok, please ignore this. I ended up using map[string]interface{} as well since more convenient.